tests/Aspire.Hosting.Tests/DistributedApplicationTests.cs

```diff
@@ -666,8 +682,11 @@ public async Task ProxylessAndProxiedEndpointBothWorkOnSameResource()
 
         var token = new CancellationTokenSource(TimeSpan.FromMinutes(1)).Token;
 
+        await Task.Delay(5000);
```

Comment:
@ReubenBond I added a delay here because the first `client.GetStringAsync($"{httpEndPoint}urls"` returned only the `http` endpoint. But the second invocation got both `http`, and `https`. Does this need to wait for all the endpoints to get allocated?

Reply:
Should the API surface something (event?) for the endpoints getting allocated?
For this test, we could either use `IDistributedApplicationLifecycleHook.AfterEndpointsAllocatedAsync` or retry a few times in a loop.

Reply:
The tests have an existing class for the hook which I tried to use here - https://gist.github.com/radical/7ef257229178c36f6b72ffd09f7e0b79#file-test-cs-L22-L27

But running the test with CWLs added to print the results from `/urls`:
```
Got urls from http://localhost:1234/urls: ["http://localhost:1234"]

Got urls from https://localhost:57048/urls: ["http://localhost:1234","https://localhost:57051"]
```

The first call is missing the `https` endpoint. Did the `AfterEndpointsAllocatedAsync` get invoked too early? Or am I missing something else?
@ReubenBond @DamianEdwards